### PR TITLE
[SPARK-15637][SPARK-15931][SPARKR] Fix R masked functions checks

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Because of the fix in SPARK-15684, this exclusion is no longer necessary.
## How was this patch tested?

unit tests

@shivaram

### CODE REVIEW

**Comment by shivaram** [comment]
LGTM. Just to confirm your local tests pass with R version > 3.2 ?

**Comment by felixcheung** [comment]
great point - @wangmiao1981 it looks like `startsWith` and `endsWith` are still masked on R 3.3?
re: PR #13476

**Comment by JoshRosen** [comment]
I'm still seeing errors after this change in my environment:

```
Failed -------------------------------------------------------------------------
1. Failure: Check masked functions (@test_context.R#31) ------------------------
length(maskedBySparkR) not equal to length(namesOfMasked).
1/1 mismatches
[1] 22 - 20 == 2


2. Failure: Check masked functions (@test_context.R#32) ------------------------
sort(maskedBySparkR) not equal to sort(namesOfMasked).
Lengths differ: 22 vs 20


DONE ===========================================================================
```

**Comment by JoshRosen** [comment]
For reference, I was using 

```
R version 3.3.0 (2016-05-03) -- "Supposedly Educational"
Copyright (C) 2016 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)
```

**Comment by shivaram** [comment]
@liancheng we can use this PR to also address https://issues.apache.org/jira/browse/SPARK-15931

**Comment by liancheng** [comment]
@shivaram True.

@felixcheung Could you please also add SPARK-15931 to the PR title if this PR also targets that one? Thanks.

**Comment by felixcheung** [comment]
FYI, this is what I see running on R 3.3:
```
1. Error: string operators (@test_sparkSQL.R#1163) -----------------------------
unable to find an inherited method for function ‘startsWith’ for signature ‘"character"’
1: expect_true(startsWith("Hello World", "Hello")) at /opt/spark-2.0.0-bin-hadoop2.6/R/lib/SparkR/tests/testthat/test_sparkSQL.R:1163
2: expect(identical(as.vector(object), TRUE), sprintf("%s isn't true.", lab), info = info)
3: as.expectation(exp, ..., srcref = srcref)
4: identical(as.vector(object), TRUE)
5: as.vector(object)
6: startsWith("Hello World", "Hello")
7: (function (classes, fdef, mtable)
   {
       methods <- .findInheritedMethods(classes, fdef, mtable)
       if (length(methods) == 1L)
           return(methods[[1L]])
       else if (length(methods) == 0L) {
           cnames <- paste0("\"", vapply(classes, as.character, ""), "\"", collapse = ", ")
           stop(gettextf("unable to find an inherited method for function %s for signature %s",
               sQuote(fdef@generic), sQuote(cnames)), domain = NA)
       }
       else stop("Internal error in finding inherited methods; didn't return a unique method",
           domain = NA)
   })(list("character"), structure(function (x, prefix)
   {
       standardGeneric("startsWith")
   }, generic = structure("startsWith", package = "SparkR"), package = "SparkR", group = list(), valueClass = character(0), signature = c("x",
   "prefix"), default = `\001NULL\001`, skeleton = (function (x, prefix)
   stop("invalid call in method dispatch to 'startsWith' (no default method)", domain = NA))(x,
       prefix), class = structure("nonstandardGenericFunction", package = "methods")),
       <environment>)
8: stop(gettextf("unable to find an inherited method for function %s for signature %s",
       sQuote(fdef@generic), sQuote(cnames)), domain = NA)
```

I will double check if there is indeed a problem with `startsWith("Hello World", "Hello")` when SparkR is loaded, or the tests are not up-to-date on what's being masked.

**Comment by felixcheung** [comment]
verified test passed and `startsWith` working
```

 Welcome to
    ____              __
   / __/__  ___ _____/ /__
  _\ \/ _ \/ _ `/ __/  '_/
 /___/ .__/\_,_/_/ /_/\_\   version  2.0.0-SNAPSHOT
    /_/


 Spark context is available as sc, SQL context is available as sqlContext
> startsWith("Hello me", "Hello")
[1] TRUE
```

**Comment by felixcheung** [comment]
changed test output to show a diff, for example:
```
Failed -------------------------------------------------------------------------
1. Failure: Check masked functions (@test_context.R#35) ------------------------
length(maskedBySparkR) not equal to length(namesOfMasked).
1/1 mismatches
[1] 22 - 21 == 1


2. Failure: Check masked functions (@test_context.R#40) ------------------------
sort(maskedBySparkR, na.last = TRUE) not equal to sort(namesOfMasked, na.last = TRUE).
10/22 mismatches
x[13]: "rank"
y[13]: "rbind"

x[14]: "rbind"
y[14]: "sample"

x[15]: "sample"
y[15]: "sd"

x[16]: "sd"
y[16]: "startsWith"

x[17]: "startsWith"
y[17]: "subset"

x[18]: "rank"
y[18]: "rbind"

x[19]: "rbind"
y[19]: "sample"

x[20]: "sample"
y[20]: "sd"

x[21]: "sd"
y[21]: "startsWith"

x[22]: "startsWith"
y[22]: "subset"
```

File: R/pkg/inst/tests/testthat/test_context.R [hunk 1]
```diff
@@ -19,21 +19,25 @@ context("test functions in sparkR.R")
 
 test_that("Check masked functions", {
   # Check that we are not masking any new function from base, stats, testthat unexpectedly
-  masked <- conflicts(detail = TRUE)$`package:SparkR`
-  expect_true("describe" %in% masked)  # only when with testthat..
-  func <- lapply(masked, function(x) { capture.output(showMethods(x))[[1]] })
-  funcSparkROrEmpty <- grepl("\\(package SparkR\\)$|^$", func)
-  maskedBySparkR <- masked[funcSparkROrEmpty]
+  # This list should not change!
```
Comment (shivaram) [inline R/pkg/inst/tests/testthat/test_context.R]: I'm not sure I follow what this comment means

**Comment by shivaram** [comment]
Thanks @felixcheung - @JoshRosen / @liancheng can you also test this PR with R 3.3.0 before we merge ?

File: R/pkg/inst/tests/testthat/test_context.R (hunk 1 above)
Comment (felixcheung) [inline R/pkg/inst/tests/testthat/test_context.R]: just want to make sure people are aware that this list of names shouldn't be added to without considering what it means? if there's a better way to convey that let me know?

Comment (shivaram) [inline R/pkg/inst/tests/testthat/test_context.R]: Yeah thats a good message to have. We can frame it as something like
```
NOTE: We should avoid adding entries to this list as masked functions make it hard for users to use base R functions.
```

Comment (felixcheung) [inline R/pkg/inst/tests/testthat/test_context.R]: 👍  right, and contact you? :)

**Comment by JoshRosen** [comment]
This passes tests for me on R 3.3.0 on OS X.

**Comment by dongjoon-hyun** [comment]
This passes for me, too. Thank you, @felixcheung .

**Comment by felixcheung** [comment]
Thanks for testing. Updated as per the last comment.

**Comment by liancheng** [comment]
Works for me too, passes tests on R 3.3.0 on OS X 10.10. Thanks!

**Comment by shivaram** [comment]
LGTM. Thanks all. I will merge this after Jenkins passes

**Pull request closed without merging**